File: slidingwindow/209-minimum-size-subarray-sum/MinSubArrayLen.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left = 0
        sum = 0
        minLen = float('inf')
        for right in range(len(nums)):
            sum+=nums[right]
            while sum>=target:
                minLen = min(minLen, right-left + 1)
                sum-=nums[left]
                left+=1
        if minLen== float('inf'):
            return 0
        else:
            return minLen

File: slidingwindow/209-minimum-size-subarray-sum/MinSubArrayLen.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int left=0, sum = 0;
        int minLen = Integer.MAX_VALUE;
        int n = nums.length;
        for(int right=0;right<n;right++){
            sum+=nums[right];
            while(sum>=target){
                minLen = Math.min(minLen, right-left+1);
                sum-=nums[left++];
            }
        }
        if(minLen==Integer.MAX_VALUE){
            return 0;
        }else{
            return minLen;
        }

    }
}